gradle/shared-with-buildSrc/mirrors.gradle.kts

```diff
@@ -14,34 +14,71 @@
  * limitations under the License.
  */
 
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
-val repositoryMirrors = findMirrorUrls()
-rootProject.extensions.extraProperties.set("repositoryMirrors", repositoryMirrors)
 
-val gradleMirrorUrl = repositoryMirrors["gradle"]
-if (gradleMirrorUrl != null) {
-    project.allprojects {
-        buildscript.repositories {
-            maven {
-                name = "gradle-mirror"
-                url = uri(gradleMirrorUrl)
-            }
-        }
-        repositories {
-            maven {
-                name = "gradle-mirror"
-                url = uri(gradleMirrorUrl)
+val originalUrls: Map<String, String> = mapOf(
+    "jcenter" to "https://jcenter.bintray.com/",
+    "mavencentral" to "https://repo.maven.apache.org/maven2/",
+    "google" to "https://dl.google.com/dl/android/maven2/",
+    "gradle" to "https://repo.gradle.org/gradle/repo",
+    "gradleplugins" to "https://plugins.gradle.org/m2",
+    "gradlejavascript" to "https://repo.gradle.org/gradle/javascript-public",
+    "gradle-libs" to "https://repo.gradle.org/gradle/libs",
+    "gradle-releases" to "https://repo.gradle.org/gradle/libs-releases",
+    "gradle-snapshots" to "https://repo.gradle.org/gradle/libs-snapshots",
+    "kotlindev" to "https://dl.bintray.com/kotlin/kotlin-dev/",
+    "kotlineap" to "https://dl.bintray.com/kotlin/kotlin-eap/"
+)
+
+val mirrorUrls: Map<String, String> =
+    System.getenv("REPO_MIRROR_URLS")?.split(',')?.associate { nameToUrl ->
+        val (name, url) = nameToUrl.split(':', limit = 2)
+        name to url
+    } ?: emptyMap()
+
+fun withMirrors(handler: RepositoryHandler) {
+    if ("CI" !in System.getenv()) {
+        return
+    }
+    handler.all {
+        if (this is MavenArtifactRepository) {
```

Comment:
Should we do `IvyArtifactRepository`, too? We use an ivy repository for downloading the distribution in , but I think we don't have a mirror for it, see https://github.com/gradle/gradle/blob/1e4bb4edaaa74cde83ef0063e0b40f34dda1ae2f/subprojects/distributions/binary-compatibility.gradle#L31-L40

I wouldn't do anything about ivy repositories now, just commenting that we also have some ivy repositories in our build.

Reply:
I wasn't aware of this repository. I'll mirror it as well, because sooner or later we'll need it.

Reply:
That is `services.gradle.org`. I am not sure we want to mirror it, given that it is backed by S3/Cloudflare and not jcenter. It also is not really a repository, it is the store where we have our distributions.

Reply:
ok, then let's do it when we really see any problems.